itest/lnd_funding_test.go

```diff
@@ -733,26 +746,62 @@ func testBatchChanFunding(ht *lntest.HarnessTest) {
 		Channels: []*lnrpc.BatchOpenChannel{{
 			NodePubkey:         bob.PubKey[:],
 			LocalFundingAmount: 100_000,
+			BaseFee:            1337,
+			UseBaseFee:         true,
 		}, {
 			NodePubkey:         carol.PubKey[:],
 			LocalFundingAmount: 100_000,
+			FeeRate:            1337,
+			UseFeeRate:         true,
 		}, {
 			NodePubkey:         dave.PubKey[:],
 			LocalFundingAmount: 100_000,
+			BaseFee:            1337,
+			UseBaseFee:         true,
+			FeeRate:            1337,
+			UseFeeRate:         true,
+		}, {
+			NodePubkey:         eve.PubKey[:],
+			LocalFundingAmount: 100_000,
+			Private:            true,
+			ZeroConf:           true,
+			CommitmentType:     lnrpc.CommitmentType_ANCHORS,
 		}},
 	}
 
+	// Check that batch opening fails due to the minchansize requirement.
 	err := alice.RPC.BatchOpenChannelAssertErr(batchReq)
 	require.Contains(ht, err.Error(), "initial negotiation failed")
 
 	// Let's fix the minimum amount for Alice now and try again.
 	batchReq.Channels[1].LocalFundingAmount = 200_000
+
+	// Set up a ChannelAcceptor for Eve to accept a zero-conf opening from
+	// Alice.
+	acceptStream, cancel := eve.RPC.ChannelAcceptor()
+	go acceptChannel(ht.T, true, acceptStream)
+
+	// Batch-open all channels.
 	batchResp := alice.RPC.BatchOpenChannel(batchReq)
-	require.Len(ht, batchResp.PendingChannels, 3)
+	require.Len(ht, batchResp.PendingChannels, 4)
 
 	txHash, err := chainhash.NewHash(batchResp.PendingChannels[0].Txid)
 	require.NoError(ht, err)
 
+	// Remove the ChannelAcceptor.
+	cancel()
+
+	// Ensure that Alice can send funds to Eve via the zero-conf channel
+	// before the batch transaction was mined.
+	eveInvoiceParams := &lnrpc.Invoice{
+		Value:   int64(10_000),
+		Private: true,
+	}
+	eveInvoiceResp := eve.RPC.AddInvoice(eveInvoiceParams)
+	ht.CompletePaymentRequests(
```

Comment:
Looks like two itests failed because of this, so this is probably flaky. Not sure if we need to wait for the channel to come online first?

Reply:
Thank you, locally it worked. I am waiting now until the channel is seen by the network.